clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/MonitorTitusScalingPolicy.java

```diff
@@ -0,0 +1,125 @@
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.netflix.spinnaker.clouddriver.event.EventMetadata;
+import com.netflix.spinnaker.clouddriver.saga.SagaCommand;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.models.Saga;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.titus.TitusClientProvider;
+import com.netflix.spinnaker.clouddriver.titus.client.TitusAutoscalingClient;
+import com.netflix.spinnaker.clouddriver.titus.credentials.NetflixTitusCredentials;
+import com.netflix.spinnaker.clouddriver.titus.deploy.events.TitusScalingPolicyModified;
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.kork.exceptions.IntegrationException;
+import com.netflix.spinnaker.kork.exceptions.UserException;
+import com.netflix.titus.grpc.protogen.ScalingPolicyResult;
+import com.netflix.titus.grpc.protogen.ScalingPolicyStatus;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MonitorTitusScalingPolicy
+    implements SagaAction<MonitorTitusScalingPolicy.MonitorTitusScalingPolicyCommand> {
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final TitusClientProvider titusClientProvider;
+  private final RetrySupport retrySupport;
+
+  @Autowired
+  public MonitorTitusScalingPolicy(
+      AccountCredentialsProvider accountCredentialsProvider,
+      TitusClientProvider titusClientProvider,
+      RetrySupport retrySupport) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.titusClientProvider = titusClientProvider;
+    this.retrySupport = retrySupport;
+  }
+
+  @NotNull
+  @Override
+  public Result apply(
+      @NotNull MonitorTitusScalingPolicy.MonitorTitusScalingPolicyCommand command,
+      @NotNull Saga saga) {
+    TitusScalingPolicyModified event = saga.getEvent(TitusScalingPolicyModified.class);
+
+    saga.log(
+        "Monitored Titus Scaling Policy %s:%s:%s (scalingPolicyId: %s",
+        event.getAccount(), event.getRegion(), event.getJobId(), event.getScalingPolicyId());
+
+    AccountCredentials accountCredentials =
+        accountCredentialsProvider.getCredentials(event.getAccount());
+
+    TitusAutoscalingClient titusClient =
+        titusClientProvider.getTitusAutoscalingClient(
+            (NetflixTitusCredentials) accountCredentials, event.getRegion());
+
+    if (titusClient == null) {
+      throw new UserException("Autoscaling is not supported for this account/region");
+    }
+
+    // make sure the new policy was applied
+    retrySupport.retry(
```

Comment:
I've not done it yet but in theory, the saga conversation would let us drop this `retrySupport` usage.

Reply:
I think a few network retries are OK.  We could potentially drop the total number of retries down to 5 though - if it's failing after 5 times due to a retryable exception it may be best to just give up and let the task run again at a later time.